src/index.js

```diff
@@ -31,19 +31,44 @@ const closeArgsToError = (code, signal) => {
   return null
 }
 
+const concatBuffer = (buffer) => {
+  if (!(buffer instanceof Array)) {
+    // throw new Error('Expected buffer to be an Array')
+    return
+  } else if (!buffer.length) {
+    return
+  }
+  let result
+  if (typeof buffer[0] === 'string') {
+    result = buffer.join('')
+  } else if (buffer[0] instanceof Buffer) {
+    result = Buffer.concat(buffer)
+  } else {
+    // throw new Error('Unexpected buffer type')
+    return
+  }
+  return result
+}
+
 export default (cmd, args, options = {}) => new Promise((resolve, reject) => {
   const proc = crossSpawn(cmd, args, options)
   let stdout = null
   let stderr = null
   const [ignoreStdout, ignoreStderr] = parseStdioOption(options.stdio)
   if (!ignoreStdout) {
     stdout = []
+    if (options.encoding) {
+      proc.stdout.setEncoding(options.encoding)
```

Comment:
I tried to avoid `if` here and do `setEncoding(options.encoding || null)` as [it said](https://nodejs.org/api/stream.html#stream_readable_setencoding_encoding):
>Encoding can be disabled by calling `readable.setEncoding(null)`. 
but it didn't work as expected in testing, `setEncoding(options.encoding || null)` with `options.encoding` definitely being `undefined` still set the encoding to `utf8` for some reason.